Xamarin.Forms.Platform.Android/Renderers/ListViewAdapter.cs

```diff
@@ -474,25 +501,19 @@ void DisposeCells()
 				if (layout == null || layout.IsDisposed())
 					continue;
 
-				var renderedView = layout?.GetChildAt(0);
+				if (_layoutsCreated.Contains(layout))
+					_layoutsCreated.Remove(layout);
 
-				var element = (renderedView as INativeElementView)?.Element;
-
-				var view = (element as ViewCell)?.View;
-
-				if (view != null)
-				{
-					var renderer = Platform.GetRenderer(view);
-
-					if (renderer == renderedView)
-						element.ClearValue(Platform.RendererProperty);
-
-					renderer?.Dispose();
-					renderer = null;
-				}
+				DisposeOfConditionalFocusLayout(layout);
+			}
 
-				renderedView?.Dispose();
-				renderedView = null;
+			// Having just this loop would probably be sufficient but
```

Comment:
It looks like this covers everything the previous loop does (and more); let's just loop once. It'll make the code easier to understand.

Reply:
@hartez fixed